### PR TITLE
Update IonQBackend to handle circuit lists of size one

### LINKED ISSUE #200: Unable to use Qiskit features such as AmplitudeEstimation
I've been trying to play around with [this example](https://qiskit.org/documentation/finance/tutorials/00_amplitude_estimation.html) from Qiskit's docs. To submit the circuit to IonQ via Azure, I'm using [these instructions](https://docs.microsoft.com/en-us/azure/quantum/quickstart-microsoft-qiskit?pivots=platform-ionq#connect-to-the-azure-quantum-service). 
When I get to the line `ae_result = ae.estimate(problem)`, I end up getting the following error:
```
FAILURE: Can not get job id, Resubmit the qobj to get job id. Error: 'list' object has no attribute 'clbits' 
```
This is printed out in a loop until my computer runs out of RAM. If I run the code on IBM's online Quantum Lab, the code terminates with the error:
```
FAILURE: Can not get job id, Resubmit the qobj to get job id. Error: 'list' object has no attribute 'clbits' 
FAILURE: Can not get job id, Resubmit the qobj to get job id. Error: 'list' object has no attribute 'clbits' 
[...]
FAILURE: Can not get job id, Resubmit the qobj to get job id. Error: 'list' object has no attribute 'clbits' 
Traceback (most recent call last):
  File "/tmp/ipykernel_240/1492105537.py", line 1, in <module>
    ae_result = ae.estimate(problem)
  File "/opt/conda/lib/python3.8/site-packages/qiskit/algorithms/amplitude_estimators/ae.py", line 314, in estimate
    counts = self._quantum_instance.execute(circuit).get_counts()
  File "/opt/conda/lib/python3.8/site-packages/qiskit/utils/quantum_instance.py", line 774, in execute
    run_circuits(
  File "/opt/conda/lib/python3.8/site-packages/qiskit/utils/run_circuits.py", line 507, in run_circuits
    job, job_id = _safe_submit_circuits(
  File "/opt/conda/lib/python3.8/site-packages/qiskit/utils/run_circuits.py", line 694, in _safe_submit_circuits
    raise QiskitError("Max retry limit reached. Failed to submit the qobj correctly")
QiskitError: 'Max retry limit reached. Failed to submit the qobj correctly'
```

Is this the intended behavior? I'm able to run the circuit if I construct it manually via `ae.construct_circuit(problem, measurement=True)` and submit that to IonQ, but this way I can't use any of the nice post-processing provided by the wrapper.

Apologies if this isn't the right way to ask, but I have no idea if this is a Qiskit, Azure, or IonQ issue.

### PR DESCRIPTION
**Pull request by jplewa**
closes #200 

Qiskit's [`BackendV1`](https://github.com/Qiskit/qiskit-terra/blob/1ecd16a0c1ef4ada08ced1c10844702e4bf58e17/qiskit/providers/backend.py#L211) supports lists of circuits in its `run` method. This isn't implemented in azure-quantum and qiskit-ionq, which means that passing a list results in confusing exceptions. A while ago, qiskit-ionq added some changes that make it possible to handle circuit lists of length 1 and also added more explicit error messages (see https://github.com/Qiskit-Partners/qiskit-ionq/pull/71). They also have an issue for adding multi-circuit experiment support in the future (see https://github.com/Qiskit-Partners/qiskit-ionq/issues/70).

I believe it would make sense to do the same in azure-quantum, because currently some of Qiskit's features are not usable. I reported this issue in https://github.com/microsoft/qdk-python/issues/200.

~~To make Qiskit's features such as [`MaximumLikelihoodAmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.MaximumLikelihoodAmplitudeEstimation.html) usable, azure-quantum would actually need to learn how to handle multi-circuit experiments, but with this PR we can at least make [`AmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.AmplitudeEstimation.html), [`IterativeAmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.IterativeAmplitudeEstimation.html), and [`FasterAmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.FasterAmplitudeEstimation.html) usable, since they pass single-element circuit lists to the `run` method.~~

EDIT: With this PR, every amplitude estimation method from Qiskit should start working.

### CODE REVIEW

**Comment by jplewa** [comment]
> To make Qiskit's features such as [`MaximumLikelihoodAmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.MaximumLikelihoodAmplitudeEstimation.html) usable, azure-quantum would actually need to learn how to handle multi-circuit experiments, but with this PR we can at least make [`AmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.AmplitudeEstimation.html), [`IterativeAmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.IterativeAmplitudeEstimation.html), and [`FasterAmplitudeEstimation`](https://qiskit.org/documentation/stubs/qiskit.algorithms.FasterAmplitudeEstimation.html) usable, since they pass single-element circuit lists to the `run` method.

I re-verified the claim I'd made about `MaximumLikelihoodAmplitudeEstimation`. This actually isn't true! Once this PR is merged, MLAE should work due to the code changes Terra introducted in https://github.com/Qiskit/qiskit-terra/pull/6391. There's a clever bit of code that splits circuit lists into one-element batches if the backend requires that (which azure-quantum does).

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py [hunk 1]
```diff
@@ -49,6 +49,12 @@ def run(self, circuit, **kwargs):
         """Submits the given circuit to run on an IonQ target."""        
         # If the circuit was created using qiskit.assemble,
         # disassemble into QASM here
+
+        if isinstance(circuit, (list, tuple)):
+            if len(circuit) > 1:
+                raise NotImplementedError("Multi-experiment jobs are not supported!")
+            circuit = circuit[0]
+
```
Comment (guenp) [inline azure-quantum/azure/quantum/qiskit/backends/ionq.py:57]:
```suggestion
        # Some Qiskit features require passing lists of circuits, so unpack those here.
        # We currently only support single-experiment jobs.
        if isinstance(circuit, (list, tuple)):
            if len(circuit) > 1:
                raise NotImplementedError("Multi-experiment jobs are not supported!")
            circuit = circuit[0]

        # If the circuit was created using qiskit.assemble,
        # disassemble into QASM here
```

**Review by guenp — CHANGES_REQUESTED** [review]
Thanks @jplewa for fixing this issue and adding tests as well! It looks good to me but before we merge to `main` we also want to support this on the Honeywell backend, i.e. in `HoneywellBackend.run`.

**Review by guenp — APPROVED** [review]

**Pull request merged**